src/ImageRequest.js

```diff
@@ -107,17 +107,12 @@ class ImageRequest {
     return true
   }
 
-  /**
-   * Decodes the image request path associated with default
-   * image requests. Provides error handling for invalid or undefined path values.
-   * @param {Object} event - The proxied request object.
-   */
-  _parseQueryParams () {
-    let qp = this.event.queryStringParameters
-    if (!qp) {
-      qp = {}
-    }
-    return schemaParser.replaceAliases(qp)
+  normalizeQueryParams (params = {}) {
+    let normalizedParams = schemaParser.replaceAliases(params)
+
+    normalizedParams.fm = this.getAutoFormat() || normalizedParams.fm || this.originalMetadata.format
+
+    return normalizedParams
```

Comment:
This ensures we always have a `fm`. Previously, with no fm param, and no `auto=format`, it would throw an error later when `toFormat` was called.